src/EThree.ts

```diff
@@ -132,48 +172,68 @@ export default class EThree {
     async decrypt(message: ArrayBuffer, publicKey?: VirgilPublicKey): Promise<Buffer>;
     async decrypt(message: Data, publicKey?: VirgilPublicKey): Promise<Data> {
         const isMessageString = isString(message);
-        const privateKey = await this.keyLoader.loadLocalPrivateKey();
+        const privateKey = await this[_keyLoader].loadLocalPrivateKey();
         if (!privateKey) throw new RegisterRequiredError();
-        if (!publicKey) publicKey = this.toolbox.virgilCrypto.extractPublicKey(privateKey);
-        let res: Data = this.toolbox.virgilCrypto.decryptThenVerify(message, privateKey, publicKey);
+        if (!publicKey) publicKey = this.virgilCrypto.extractPublicKey(privateKey);
+        let res: Data = this.virgilCrypto.decryptThenVerify(message, privateKey, publicKey);
         if (isMessageString) return res.toString('utf8') as string;
         return res as Buffer;
     }
 
     async lookupPublicKeys(identities: string): Promise<VirgilPublicKey>;
-    async lookupPublicKeys(identities: string[]): Promise<VirgilPublicKey[]>;
-    async lookupPublicKeys(
-        identities: string[] | string,
-    ): Promise<VirgilPublicKey[] | VirgilPublicKey> {
+    async lookupPublicKeys(identities: string[]): Promise<LookupResult>;
+    async lookupPublicKeys(identities: string[] | string): Promise<LookupResult | VirgilPublicKey> {
         const argument = isArray(identities) ? identities : [identities];
-
         if (argument.length === 0) throw new EmptyArrayError('lookupPublicKeys');
+        if (hasDuplicates(argument)) throw new Error(DUPLICATE_IDENTITES);
 
-        const responses = await Promise.all(
-            argument.map(i =>
-                this.toolbox
-                    .getPublicKey(i)
-                    .catch(e => Promise.resolve(e instanceof Error ? e : new Error(e))),
-            ),
-        );
+        const cards = await this.cardManager.searchCards(argument);
 
-        if (isWithoutErrors(responses)) return isArray(identities) ? responses : responses[0];
+        let result: LookupResult = {},
+            resultWithErrors: LookupResultWithErrors = {};
```

Comment:
why not just use type `{ [identity: string]: Error }` since you only write errors in it? Then you can define `LookupResultWithErrors` in the `errors.ts` file and the code's purpose would be clearer _imho_